plugins/upower/fu-plugin-upower.c

```diff
@@ -33,6 +33,42 @@ fu_plugin_destroy (FuPlugin *plugin)
 		g_object_unref (data->display_proxy);
 }
 
+static gboolean
+fu_plugin_upower_has_device_obj (FuPlugin *plugin, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	g_autoptr(GVariant) val = NULL;
+	g_autoptr(GVariantIter) obj_array = NULL;
+	gchar *obj_path_str = NULL;
+
+	val = g_dbus_proxy_call_sync (data->upower_proxy,
+				      "EnumerateDevices",
+				      NULL,
+				      G_DBUS_CALL_FLAGS_NONE,
+				      -1,
+				      NULL,
+				      error);
+
+	// Check if D-Bus proxy reponses
+	if (val == NULL || !error )
+		return FALSE;
+
+	// Check the type of return in response to EnumerateDevices()
+	if (!g_variant_is_of_type (val, G_VARIANT_TYPE("(ao)"))){
```

Comment:
I tried to compare val with type [G_VARIANT_TYPE_OBJECT_PATH_ARRAY](https://developer.gnome.org/glib/stable/glib-GVariantType.html#G-VARIANT-TYPE-OBJECT-PATH-ARRAY:CAPS) but the return in fact is "(ao)" instead of "ao" that was pre-defined in glib2.

Given that inconsistency I use G_VARIANT_TYPE with manual format at this place.